File: core/entities/src/main/java/de/iisys/sysint/hicumes/core/entities/SuspensionType.java
package de.iisys.sysint.hicumes.core.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonView;
import de.iisys.sysint.hicumes.core.entities.jsonViews.JsonViews;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;

import javax.persistence.*;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@ToString(callSuper = true)
@XmlRootElement(name = "suspensionType")
@XmlAccessorType(XmlAccessType.FIELD)
@Table(indexes = @Index(columnList = "externalId"))
public class SuspensionType extends EntitySuperClass{

    @Column(unique = true)
    private String name;

    private String description;
    /*@JsonView({JsonViews.SchemaMapperDefault.class, JsonViews.ViewMachineOccupation.class})
    @OneToMany(mappedBy = "suspensionType", cascade = CascadeType.ALL)
    @LazyCollection(LazyCollectionOption.FALSE)
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @ToString.Exclude
    private List<TimeRecord> timeRecords = new ArrayList<>();*/

    public SuspensionType(String name, String description) {
        this.name = name;
        this.description = description;
    }
}

File: docker_compose/feinplanung/HiCuMes.Persistence/Entities/Suspensiontype.cs
using System;
using System.Collections.Generic;

namespace HiCuMes.Persistence.Entities;

public partial class Suspensiontype
{
    public long Id { get; set; }

    public DateTime? CreateDateTime { get; set; }

    public string? ExternalId { get; set; }

    public DateTime? UpdateDateTime { get; set; }

    public string? VersionNr { get; set; }

    public string? Description { get; set; }

    public string? Name { get; set; }

    public  ICollection<Timerecord> Timerecords { get; set; } = new List<Timerecord>();
}
